src/main/scala/cosim/Cospike.scala

```diff
@@ -64,10 +66,20 @@ class SpikeCosim(cfg: SpikeCosimConfig) extends BlackBox(Map(
   })
 }
 
+class SpikeCosimRegisterMemory(base: BigInt, size: BigInt) extends BlackBox(Map(
+  "BASE" -> IntParam(base),
+  "SIZE" -> IntParam(size)))
+{
+  val io = IO(new Bundle {})
+}
+
 object SpikeCosim
 {
   def apply(trace: TileTraceIO, hartid: Int, cfg: SpikeCosimConfig) = {
     val cosim = Module(new SpikeCosim(cfg))
+    for ((base, size) <- cfg.mems) {
+      val reg = Module(new SpikeCosimRegisterMemory(base, size))
```

Comment:
Is this module's DPI calls in the `initial` block guaranteed to be before the `SpikeCosim` module in this code? I'm not sure what order initial blocks are called if the `reg` module is after the `cosim` module in the code.

Reply:
The `initial` blocks can be called in any order, but all the initial blocks should be executed before any of the actual simulation verilog is evaluated.